src/Promotion/Action/UnitsFixedPricePromotionActionCommand.php

```diff
@@ -0,0 +1,169 @@
+    ) {
+        parent::__construct($adjustmentFactory);
+
+        $this->priceRangeFilter = $priceRangeFilter;
+        $this->taxonFilter = $taxonFilter;
+        $this->productFilter = $productFilter;
+        $this->distributor = $distributor;
+    }
+
+    public function execute(PromotionSubjectInterface $subject, array $configuration, PromotionInterface $promotion): bool
+    {
+        if (!$subject instanceof OrderInterface) {
+            throw new UnexpectedTypeException($subject, OrderInterface::class);
+        }
+
+        $channel = $subject->getChannel();
+        if (null === $channel) {
+            return false;
+        }
+
+        $channelCode = $channel->getCode();
+        if (!isset($configuration[$channelCode])) {
+            return false;
+        }
+
+        if (empty($configuration[$channelCode]['amount']) || empty($configuration[$channelCode]['itemsAmount'])) {
+            return false;
+        }
+        $amount = $configuration[$channelCode]['amount'];
+        $itemsAmount = $configuration[$channelCode]['itemsAmount'];
+
+        $filteredItems = $this->getFilteredItems($subject, $configuration[$channelCode]);
+        if (empty($filteredItems)) {
+            return false;
+        }
+
+        $amount = $this->getExpectedTotal($amount, $itemsAmount, $filteredItems);
+        // If the items total is already lower than expected discounted amount, return there
+        $currentTotal = $this->getEligibleItemsTotal($filteredItems);
+        if ($currentTotal <= $amount) {
+            return false;
+        }
+
+        $discountAmount = $currentTotal - $amount;
+        $units = $this->getUnits($filteredItems);
+        $unitsTotals = [];
+        foreach ($units as $unit) {
+            $unitsTotals[] = $unit->getTotal();
+        }
+
+        $splitPromotion = $this->distributor->distribute($unitsTotals, $discountAmount);
+
+        foreach ($units as $key => $unit) {
+            $this->setUnitAdjustments($unit, $splitPromotion[$key], $promotion);
+        }
+
+        return true;
+    }
+
+    private function setUnitAdjustments(OrderItemUnitInterface $unit, int $amount, PromotionInterface $promotion): void
+    {
+        $this->addAdjustmentToUnit(
+            $unit,
+            min($unit->getTotal(), $amount),
+            $promotion
+        );
+    }
+
+    /**
+     * @return iterable|OrderItemInterface[]
+     */
+    private function getFilteredItems(OrderInterface $order, array $configuration): iterable
+    {
+        $filteredItems = $this->priceRangeFilter->filter(
+            $order->getItems()->toArray(),
+            array_merge(['channel' => $order->getChannel()], $configuration)
+        );
+        $filteredItems = $this->taxonFilter->filter($filteredItems, $configuration);
+
+        return $this->productFilter->filter($filteredItems, $configuration);
+    }
+
+    private function getExpectedTotal(int $expectedAmount, int $itemsAmount, iterable $items): int
```

Comment:
I miss some comments on these methods. It's rather difficult to follow the flow of things or the intentions when no comments are supplied